File: finance-client/src/app/api/productlineManage.service.ts
import { Injectable } from '@angular/core';
import * as global from '../global';
import {GetResponseService} from './getResponse.service';
import {Observable} from 'rxjs/Observable';
const url = global.baseUrl + '/ProductlineManageController';
// 生产线匹配关系管理页面
@Injectable()
export class ProductlineManageService {
  constructor (private response: GetResponseService) {}
  // 获取未匹配上的生产线
  getUnmatchedProductline(param): Observable<any> {
    return this.response.appPost(param, url + '/getUnmatchedProductline');
  }
  // 获取生产线匹配关系表的数据
  getMatchedProductline(param): Observable<any> {
    return this.response.appPost(param, url + '/ getMatchedProductline');
  }
  // 添加生产线匹配关系
  addMatchedProductline(param): Observable<any> {
    return this.response.appPost(param, url + '/addMatchedProductline');
  }
  // 更新生产线匹配关系
  updateMatchedProductline(param): Observable<any> {
    return this.response.appPost(param, url + '/ updateMatchedProductline');
  }
  // 删除生产线匹配关系
  deleteMatchedProductline(param): Observable<any> {
    return this.response.appPost(param, url + '/deleteMatchedProductline');
  }
}

File: finance-server/hengyi-api/src/main/java/com/hengyi/service/ProductlineManageService.java
package com.hengyi.service;

import com.hengyi.domain.ProductlineMatchDomain;
import com.hengyi.util.QueryDtoBase;

import java.util.List;

public interface ProductlineManageService {
    /***
     * 获取未匹配到生产线记录数，从而便于进行分页
     * @return
     */
    Long getUnmatchedProductlineCount();
    /***
     * 获取未匹配到的生产线信息
     * @return
     */
    List<ProductlineMatchDomain> getUnmatchedProductline(QueryDtoBase queryDtoBase);

    /***
     * 获取已经匹配上的生产线匹配关系记录数
     */
    Long getMatchedProductlineCount();

    /***
     * 获取已经有的匹配关系
     * @return
     */
    List<ProductlineMatchDomain> getMatchedProductline(QueryDtoBase queryDtoBase);

    /***
     * 添加生产线匹配关系
     * @param productlineMatchDomain
     */
    void addMatchedProductline(ProductlineMatchDomain productlineMatchDomain);

    /***
     * 修改生产线匹配关系
     * @param productlineMatchDomain
     */
    void updateMatchedProductline(ProductlineMatchDomain productlineMatchDomain);

    /***
     * 删除匹配关系
     * @param productlineMatchDomain
     */
    void deleteMatchedProductline(ProductlineMatchDomain productlineMatchDomain);
}

